packages/vhd-lib/merge.js

```diff
@@ -94,11 +94,6 @@ module.exports = limitConcurrency(2)(async function merge(
         currentBlock: 0,
         mergedDataSize: 0,
       }
-
-      // finds first allocated block for the 2 following loops
-      while (mergeState.currentBlock < maxTableEntries && !childVhd.containsBlock(mergeState.currentBlock)) {
-        ++mergeState.currentBlock
-      }
```

Comment:
What do you think of writing the BAT at the same time at the merge state instead?

Maybe we'll need to increase the delay between two writes or maybe after a given number of blocks has been written.